### PR TITLE
[DEVC-590] Add retry loop for initial AT command probe to candidate modem ttys

### PR DESCRIPTION
**Pull request by benjaminaltieri**
Retries added to initial modem probe attempt in `cellmodem_probe.c`

Also removed manual sbp log routine which has been superseded by `sbp_log` from `libpiksi`

### CODE REVIEW

File: package/piksi_system_daemon/piksi_system_daemon/src/cellmodem_probe.c [hunk 1]
```diff
@@ -167,7 +167,7 @@ enum modem_type cellmodem_probe(const char *dev, sbp_zmq_pubsub_ctx_t *pubsub_ct
       return MODEM_TYPE_INVALID;
     }
     piksi_log(LOG_INFO, "Modem %s: %s", c->display, r);
```
Comment (silverjam) [inline package/piksi_system_daemon/piksi_system_daemon/src/cellmodem_probe.c]: FYI, I added a LOG_SBP flag to the loglevel parameter to handle these cases, so you can do the following to syslog and SBP:

      piksi_log(LOG_INFO|LOG_SBP, "Modem %s: %s", c->display, r);

Comment (benjaminaltieri) [reply package/piksi_system_daemon/piksi_system_daemon/src/cellmodem_probe.c]: Cool yea I saw that, is that in 1.5 too?

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.8-2-gf501bc1
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.8-2-gf501bc1/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.8-2-gf501bc1/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.8-2-gf501bc1

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.8-3-ga1f70a7
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.8-3-ga1f70a7/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.8-3-ga1f70a7/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.8-3-ga1f70a7

**Pull request merged**